contract/src/zcash.rs

```diff
@@ -41,40 +68,29 @@ impl BtcLightClient {
     }
 }
 
+struct NextWorkResult {
+    expected_bits: u32,
+    prev_block_median_time_past: u32,
+}
+
 // Reference implementation: https://github.com/zcash/zcash/blob/v6.2.0/src/pow.cpp#L20
 fn zcash_get_next_work_required(
     config: &ZcashConfig,
     block_header: &Header,
     prev_block_header: &ExtendedHeader,
     prev_block_getter: &impl BlocksGetter,
-) -> u32 {
-    use btc_types::network::ZCASH_MEDIAN_TIME_SPAN;
-
-    if let Some(pow_allow_min_difficulty_blocks_after_height) =
```

Comment:
We have require checks further down that weren’t triggered before, but now, after moving the code to the end, they might start triggering. Do we want them to trigger?